fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/EntityElytraEvents.java

```diff
@@ -0,0 +1,116 @@
+public class EntityElytraEvents {
+	/**
+	 * An event that allows listeners to grant elytra flight to living entities.
+	 * Will be called when players try to start elytra flight by pressing space in mid-air, and every tick for all living entities to check if elytra flight is still allowed.
+	 *
+	 * <p>Items that wish to grant elytra flight when worn in the chest equipment slot can simply implement {@link FabricElytraItem} instead of registering a listener.
+	 */
+	public static final Event<Grant> GRANT = EventFactory.createArrayBacked(Grant.class, listeners -> (livingEntity, damageElytra) -> {
+		for (Grant listener : listeners) {
+			if (listener.grantElytraFlight(livingEntity, damageElytra)) {
+				return true;
+			}
+		}
+
+		return false;
+	});
+
+	static {
+		GRANT.register((livingEntity, damageElytra) -> {
+			ItemStack chestStack = livingEntity.getEquippedStack(EquipmentSlot.CHEST);
+
+			if (chestStack.getItem() instanceof FabricElytraItem fabricElytraItem) {
+				return fabricElytraItem.grantElytraFlight(livingEntity, chestStack, damageElytra);
+			}
+
+			return false;
+		});
+	}
+
+	@FunctionalInterface
+	public interface Grant {
+		/**
+		 * Check if elytra flight should be granted to an entity.
+		 *
+		 * @param livingEntity the entity
+		 * @param tickElytra   false if this is just to check if elytra flight is allowed, true if the elytra should be ticked, i.e. damaged.
+		 * @return true to grant elytra flight, and cancel subsequent handlers
+		 */
+		boolean grantElytraFlight(LivingEntity livingEntity, boolean tickElytra);
+	}
+
+	/**
+	 * An interface that can be implemented on an item to enable elytra flight when it is worn in the {@link EquipmentSlot#CHEST} slot.
+	 *
+	 * <p>To disable cape rendering when this item is worn (like the vanilla elytra item), have a look at {@code LivingEntityFeatureRenderEvents}.
+	 */
+	public interface FabricElytraItem {
+		/**
+		 * Check if the item grants elytra flight to an entity.
+		 *
+		 * @param livingEntity the entity
+		 * @param chestStack   the stack currently worn in the chest slot, will always be of this item
+		 * @param tickElytra   true to tick the elytra, false to only perform the check; vanilla-like elytras can use {@link #doVanillaElytraTick} to handle ticking
+		 * @return true to grant elytra flight
+		 */
+		default boolean grantElytraFlight(LivingEntity livingEntity, ItemStack chestStack, boolean tickElytra) {
+			if (ElytraItem.isUsable(chestStack)) {
+				if (tickElytra) {
+					doVanillaElytraTick(livingEntity, chestStack);
+				}
+
+				return true;
+			}
+
+			return false;
+		}
+
+		/**
+		 * A helper to perform the default vanilla elytra tick logic: damage the elytra every 20 ticks, and send a game event every 10 ticks.
+		 */
+		default void doVanillaElytraTick(LivingEntity livingEntity, ItemStack chestStack) {
+			int nextRoll = livingEntity.getRoll() + 1;
+
+			if (!livingEntity.world.isClient && nextRoll % 10 == 0) {
+				if ((nextRoll / 10) % 2 == 0) {
```

Comment:
Given how short the logic is, mods should imo just copy/paste it and adjust it. The primary purpose of this method is to illustrate what mods are expected to do in their "elytra tick".

Reply:
IMO the less copypasting and code repetition the better.

Reply:
> IMO the less copypasting and code repetition the better.

Vanilla minecraft logic unfortunately is just a lot of hardcoding. So we do have to resort to copypasta unfortunately.

Reply:
It's more that this is a trivial 5 lines function, and I would rather have mods copy/paste the bits they need (again, we are talking about ~5 lines that basically do 2 things) than introduce two extra functions to customize this.

We don't know if mojang is going to change the body of this function, but if they do I'd rather have mods decide if they want the new behavior instead of their old custom behavior, than having to break or extend this default helper.